packages/forms/frontend/sirius-components-widget-reference/src/components/FilterableSortableList.tsx

```diff
@@ -0,0 +1,214 @@
+            target: { value },
+          } = event;
+          setFilterBarText(value);
+        }}
+        onTextClear={() => {
+          setFilterBarText('');
+        }}
+        text={filterBarText}
+        options={{
+          textFieldVariant: 'standard',
+          searchIcon: true,
+          clearTextButton: true,
+          filterButton: false,
+          closeButton: false,
+          filterEnabled: true,
+          filterBarDisplayByDefault: true,
+        }}
+      />
+      <span className={classes.title}>Selected</span>
+      <div className={classes.borderStyle} onDrop={handleDropNewItem} onDragOver={handleDragOverNewItem}>
+        <List dense component="div" role="list">
+          {items
+            .filter(({ label }) => {
+              if (filterBarText === null || filterBarText === '') {
+                return true;
+              }
+              const splitLabelWithTextToHighlight: string[] = splitText(label, filterBarText);
+              return (
+                splitLabelWithTextToHighlight.length > 1 ||
+                (splitLabelWithTextToHighlight.length === 1 &&
+                  splitLabelWithTextToHighlight[0].toLocaleLowerCase() === filterBarText.toLocaleLowerCase())
+              );
+            })
+            .map(({ id, kind, label, iconURL }, index) => {
+              const labelId = `transfer-list-item-${id}-label`;
+              const selected = selectedItems.some((entry) => entry.id === id);
+              const hover = isHovered === id;
+              return (
+                <ListItem
+                  key={id}
+                  role="listitem"
+                  className={selected ? classes.selected : classes.selectable}
+                  onDragOver={(event) => handleDragOverOrder(event, index)}
+                  onDragEnd={handleDragItemEnd}
+                  onClick={(event) => onClick(event, { id, kind, label, iconURL })}
+                  onMouseEnter={() => handleMouseEnter(id)}
+                  onMouseLeave={handleMouseLeave}>
+                  {hover ? (
+                    <ListItemIcon draggable onDragStart={() => handleDragStartOrder(id)} onDragEnd={handleDragEndOrder}>
+                      <DragHandleIcon />
+                    </ListItemIcon>
+                  ) : null}
+                  <ListItemIcon draggable onDragStart={() => handleDragItemStart(id)}>
+                    {iconURL ? <img width="16" height="16" alt={''} src={httpOrigin + iconURL} /> : null}
+                  </ListItemIcon>
+                  <ListItemText
+                    id={labelId}
+                    draggable
+                    onDragStart={() => handleDragItemStart(id)}
+                    primary={<Typography>{highlightText(label, filterBarText, classes)}</Typography>}
+                    classes={{ primary: selected ? classes.selectedLabel : '' }}
+                  />
+                </ListItem>
+              );
+            })}
+          <ListItem />
```

Comment:
What is the purpose of this isolated `<ListItem />`?

Reply:
A mistake.